src/test_l1_dsAudio.c

```diff
@@ -3550,45 +4516,56 @@ void test_l1_dsAudio_negative_dsGetMS12AudioProfile(void) {
 
-	dsError_t result;
-	intptr_t handle[NUM_OF_PORTS];
-	char currentProfile[MAX_PROFILE_NAME_LEN];
+        dsError_t result;
+        intptr_t* handle = NULL;
+        char currentProfile[MAX_PROFILE_NAME_LEN];
 
-	// Step 01: Attempt to get MS12 Audio Profile without initializing
-	result = dsGetMS12AudioProfile(-1, currentProfile);
-	CHECK_FOR_EXTENDED_ERROR_CODE( result, dsERR_NOT_INITIALIZED, dsERR_INVALID_PARAM );
-	
-	// Step 02: Initialize audio ports
-	result = dsAudioPortInit();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // allocate memory for the handles
+        handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
+        if(handle == NULL)
+        {
+                UT_LOG_ERROR("Failed to allcoate memory for handlers");
+                return;
+        }
 
-	// Step 03: Attempt to get MS12 Audio Profile using an invalid handle
-	result = dsGetMS12AudioProfile(handle[0], currentProfile);
-	UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
+        // Step 01: Attempt to get MS12 Audio Profile without initializing
+        result = dsGetMS12AudioProfile(-1, currentProfile);
+        CHECK_FOR_EXTENDED_ERROR_CODE( result, dsERR_NOT_INITIALIZED, dsERR_INVALID_PARAM );
 
-	// Step 04: Get the port handle for all supported audio ports
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+        // Step 02: Initialize audio ports
+        result = dsAudioPortInit();
+        UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-		// Step 05: Attempt to get MS12 Audio Profile with a null pointer
-		if(handle[i] == dsAUDIOPORT_TYPE_SPEAKER) {
-			result = dsGetMS12AudioProfile(handle[i], NULL);
-			UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
-		}
-	}
+        // Step 03: Attempt to get MS12 Audio Profile using an invalid handle
+        result = dsGetMS12AudioProfile(*handle, currentProfile);
+        UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
 
-	// Step 06: Terminate audio ports
-	result = dsAudioPortTerm();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // Step 04: Get the port handle for all supported audio ports
+        for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+                result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
+                UT_ASSERT_EQUAL(result, dsERR_NONE);
+                UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
-	// Step 07: Attempt to get MS12 Audio Profile after termination
-	result = dsGetMS12AudioProfile(handle[0], currentProfile);
-	CHECK_FOR_EXTENDED_ERROR_CODE( result, dsERR_NOT_INITIALIZED, dsERR_INVALID_PARAM );
-	
-	// Logging at the end
-	UT_LOG("\n Out %s\n", __FUNCTION__);
+                // Step 05: Attempt to get MS12 Audio Profile with a null pointer
+                if(handle[i] == gDSAudioPortConfiguration[0].typeid ) {
```

Comment:
comparing handle with typeid is not correct

Reply:
done